File: JavaBankSystem/src/data/DataStore.java
package data;

import java.util.HashMap;

import model.User;

public class DataStore {
    private static HashMap<String, User> users = new HashMap<>();

    public static HashMap<String, User> getUsers() {
        return users;
    }

    public static void clear() {
        users.clear();
    }

    public static boolean existsUser(String username) {
    return users.containsKey(username);
}
}

File: CppBankSystem/src/data/DataStore.cpp
#include "DataStore.h"

std::unordered_map<std::string, User> DataStore::users;

std::unordered_map<std::string, User>& DataStore::getUsers() {
    return users;
}

void DataStore::clear() {
    users.clear();
}

bool DataStore::existsUser(const std::string& username) {
    return users.find(username) != users.end();
}